### ISSUE TITLE
[bug] Phase 1: negative cross-entropy, all-zero stationary distribution, and an inverted Cohen's d

### ISSUE THREAD

**Issue by thejesh23** [rohitg00/ai-engineering-from-scratch#343]
## Where

- Phase / lesson: Phase 1 · 13-numerical-stability, 22-stochastic-processes, 15-statistics-for-ml
- Files: `code/numerical.py`, `code/stochastic.py`, `code/statistics.py`

## What's wrong

Three wrong-math defects in Phase 1. Two of them print visibly wrong numbers; the third contradicts the lesson's own docs.

**1. `13-numerical-stability` — `binary_cross_entropy_stable` returns *negative* cross-entropy.**
The function takes a logit, so it should compute `softplus(x) - y*x`. With `max_val = max(0, -logit)`, the first term is the stable form of `log(1+e^-x)`, i.e. `softplus(-x)`, not `softplus(x)`. Since `softplus(-x) = softplus(x) - x`, every result is short by exactly the logit:

| logit | y | `naive(y, sigmoid(x))` | `stable(y, x)` |
|---|---|---|---|
| 0.5 | 1 | 0.474077 | **-0.025923** |
| 2.0 | 1 | 0.126928 | **-1.873072** |
| 6.0 | 1 | 0.002476 | **-5.997524** |

A cross-entropy of -5.9975 is impossible. For `y=0` it silently returns the other label's loss. No demo calls the function, which is why this went unnoticed.

**2. `22-stochastic-processes` — the stationary distribution prints `0.0000` for every state.**
`stationary_distribution` clips the lambda=1 eigenvector at 0. LAPACK's eigenvector sign is arbitrary, and for this lesson's weather matrix it comes back all-negative, so clipping zeroes every entry, the sum is 0, the `if total > 0` guard skips normalisation, and the all-zero vector is returned:

```
Stationary distribution (analytical):
  Sunny: 0.0000   Rainy: 0.0000   Cloudy: 0.0000
Empirical distribution (100000 steps, start=Sunny):
  Sunny: 0.5469   Rainy: 0.1826   Cloudy: 0.2705
```

`docs/en.md:94` promises roughly `[0.53, 0.18, 0.29]`, and the convergence check below it never converges (max error stays ~0.55). Perron-Frobenius guarantees the vector is single-signed, so `abs` is the correct normalisation — which is what `docs/en.md` itself shows.

**3. `15-statistics-for-ml` — `cohens_d` sign is inverted.**
`docs/en.md:301` defines `Cohen's d = (mean_1 - mean_2) / pooled_std` and the sibling `t_statistic_two_sample` returns `(m1 - m2) / se`, but `cohens_d` returns `(m2 - m1) / pooled`. The demo prints both for the same pair on adjacent lines, so the output reads `t-statistic: -2.4599` then `Cohen's d: 0.6351`. Magnitude is right and `interpret_cohens_d` uses `abs()`, so no verdict changes — but the file contradicts itself and the docs.

## Reproduce

1. `python3 -c "import numerical; print(numerical.binary_cross_entropy_stable(1.0, 6.0))"` → `-5.9975`, should be `0.0025`.
2. `python3 phases/01-math-foundations/22-stochastic-processes/code/stochastic.py` → DEMO 2 prints all zeros.
3. `python3 phases/01-math-foundations/15-statistics-for-ml/code/statistics.py` → t-statistic and Cohen's d print with opposite signs.

## Environment

- OS: macOS 15
- Python 3.9.6, numpy 2.0.2
- How you ran it: local

I have one-line fixes for all three and will open a PR referencing this issue.
